### DISCUSSION TITLE
How to change header and footer of quarto output by including a tex-file

### DISCUSSION THREAD

**Discussion by MasterGPT9821** [quarto-dev/quarto-cli#8418] [Q&A]
How to change header and footer of quarto output by including a tex-file

Here ist the beginning of my qmd-file in RStudio
```
---
title: "title"
subtitle: "subtitle"
author: |
  | "author 1"
  | "author 2"
institute: | 
  | university
  | institute
date: today
lang: de
format:
  pdf:
    documentclass: scrbook
    number-sections: true
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    include-in-header:
      - header_footer.tex
fontsize: 11pt
linestretch: 1.5
geometry: left = 25mm, right = 25mm, top = 30mm, bottom = 30mm
bibliography: bosch.bib
execute: 
  echo: false
  warning: false
knitr: 
  opts_chunk:
    message: false
editor: visual
---
```

Here is my including tex file
```
\usepackage{scrlayer-scrpage}
\pagestyle{scrheadings}

\ohead{\headmark} % Überschrift 1 im äußeren Kopf
\ifoot{\pagemark} % Seitenzahl in der Mitte des Fußes
\setheadsepline{0.5pt} % Linie unter der Kopfzeile
\setfootsepline{0.5pt} % Linie über der Fußzeile
```

**Comment by mcanouil** [comment]
Please format your post properly for it to be legible.
Also, please add your question in the body of your post, not only in title.
Thank you!

See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

**Reply by MasterGPT9821** [reply]
i have just upadeted it

**Reply by mcanouil** [reply]
Thanks!

Why to you have? What are you trying to do with this which highly irregular?

```yaml
author: |
  | "author 1"
  | "author 2"
institute: | 
  | university
  | institute
```

**Reply by MasterGPT9821** [reply]
I would like the authors to be written among themselves
I would also like the name of the institute to appear below the name of the university

That would have been my second question

**Reply by mcanouil** [reply]
Did you look at the authors/affiliations schema? <https://quarto.org/docs/journals/authors.html>
You might need to look at partials to customise the format: <https://quarto.org/docs/journals/templates.html#latex-partials>.

Take a look at the extension [Quarto Titlepages](https://github.com/nmfs-opensci/quarto_titlepages) and all the [journals format extension](https://quarto.org/docs/extensions/listing-journals.html)

Regarding your LaTeX question, you can use `keep-tex: true` to your YAML frontmatter to see how your code behaves in the context of `scrbook` which comes with some benefits/limitations described in the [CTAN page documentation](https://ctan.org/pkg/scrbook).
Then you can see where to include the code and which code if yours is not correct (https://quarto.org/docs/reference/formats/pdf.html#includes).

Side note:

```yaml
knitr: 
  opts_chunk:
    message: false
```

is a bit overkill since you are already using `execute`:
```yaml
execute:
 message: false
```

**Comment by MasterGPT9821** [comment]
Can't you give me short answer in the code sequence how can i fix this?

**Reply by mcanouil** [reply]
I don't have the LaTeX answer (note that it's really not about Quarto here-, it's LaTeX).
To provide you with one, I would have to look on the Internet and to do exactly what I am suggesting you to do with looking at the LaTeX code produced and see what and how to include the required LaTeX code needed to get the results you want.

Another side note, you are using `documentclass: scrbook` why not use a Quarto book project? <https://quarto.org/docs/reference/projects/books.html>.

**Reply by MasterGPT9821** [reply]
`documentclass: scrbook` is better for scientific work. i like `scrbook`'s formatting better

**Reply by mcanouil** [reply]
I am not talking about changing the PDF class, but to use Book project which is designed to work with books.

Follow the [Creating a Book guide](https://quarto.org/docs/books/) and replace in `_quarto.yml`:  `documentclass: scrreprt` with `documentclass: scrbook`.
Then use all the Book features which you don't have access to, since you are not using the Book project type.

**Reply by MasterGPT9821** [reply]
i'm already using the `documentclass: scrbook` ...

**Reply by mcanouil** [reply]
I know you are using `documentclass: scrbook`. That is not the point.
I am talking about the project type.
Quarto offers several project types which comes with features.
There is a Book project type designed to create books.
Take a look at the link I shared.

**Reply by MasterGPT9821** [reply]
ok. now i understand. thank you

**Comment by MasterGPT9821** [comment]
I fixed the issue with the header and footer.
keep-tex is helpful but wasn't necesarrly needed.
the problem was, that in `scrbook` (also in `book`) the sides of chapters have no header and no footer.
If there are several pages after the chapter page, the header and footer are inserted.
I have adapted the tex-code so that the `chapter` is in the `header`.

Here is my latex code:
```latex
\usepackage{scrlayer-scrpage}
\pagestyle{scrheadings}

\ohead{\headmark}
\automark[chapter]{chapter}
\setheadsepline{0.4pt}
\setfootsepline{0.4pt}
```

Here is my Quarto code of a chapter:
```qmd
{{< pagebreak >}}

test

{{< pagebreak >}}

test

{{< pagebreak >}}

## test
test
```

Here is a picture of the output.

![grafik](https://github.com/quarto-dev/quarto-cli/assets/157615015/1b40812c-0157-4e01-9401-e7b4d34725d4)

**Comment by MasterGPT9821** [comment]
Now it's about solving the problem with the authors and the institution...

**Reply by mcanouil** [reply]
That's the partials as suggested previously (https://github.com/quarto-dev/quarto-cli/discussions/8418#discussioncomment-8234419)

https://quarto.org/docs/journals/templates.html#template-partials
